protobuf_definitions/message_formats.proto

```diff
@@ -0,0 +1,319 @@
+/**
+ * Common messages
+ *
+ * These are used for logging as well as building requests and responses.
+ */
+
+syntax = "proto3";
+
+package blueye.protocol;
+option csharp_namespace = "Blueye.Protocol.Protobuf";
+
+/**
+ * Motion input from client.
+ *
+ * Used to indicate the desired motion in each direction.
+ * Typically these values map to the left and right joystick for motion,
+ * and the left and right trigger for the slow and boost modifiers.
+ */
+message MotionInput {
+  float surge = 1; // Forward (positive) and backwards (negative) movement. (-1..1).
+  float sway = 2; // Right (positive) and left (negative) lateral movement (-1..1).
+  float heave = 3; // Descend (positive) and ascend (negative) movement (-1..1).
+  float yaw = 4; // Left (positive) and right (negative) movement (-1..1).
+  float slow = 5; // Multiplier used to reduce the speed of the motion (0..1).
+  float boost = 6; // Multiplier used to increase the speed of the motion (0..1).
+}
+
+message UserModifiers {
+  float surge_modifier = 1;
+  float sway_modifier = 2;
+  float heave_modifier = 3;
+  float yaw_modifier = 4;
+  float jerk_modifier = 5;
+}
+
+message Lights {
+  uint32 percentage = 1;
+}
+
+message LatLongPosition {
+  float latitude = 1;
+  float longitude = 2;
+}
+
+message ConnectionDuration {
+  int64 connection_duration = 1;
+}
+
+message AutoHeadingState {
+  bool enabled = 1;
+}
+
+message AutoDepthState {
+  bool enabled = 1;
+}
+
+message SystemTime {
+  string unix_timestamp = 1;
+}
+
+message ErrorFlags {
+  bool missing_ack_from_pmu = 1;
+  bool pmu_communication_timeout = 2;
+  bool depth_sensor_read_error = 3;
+  bool depth_sensor_spike = 4;
+  bool inner_pressure_sensor_read_error = 5;
+  bool inner_pressure_sensor_spike = 6;
+  bool invalid_compass_calibration = 7;
+  bool invalid_tilt_calibration = 8;
+}
+
+message DroneInfo {
+  string serial_number = 1;
+  string blunux_version = 2;
+}
+
+message RecordingState {
+  bool is_recording = 1;
+}
+
+message TiltSpeed {
+  float speed = 1;
+}
+
+message WaterDensity {
+  int32 density = 1;
+}
+
+/**
+ * Duration of the currently running video recording
+ * -1 if no running recording
+ */
```

Comment:
I think it's better to be explicit with the recording state, instead of encoding it as arbitrary values (-1 in this case).

Reply:
Fixed in ebfbcb5c4873eb3a49e23fe24a18e98620d00ed0